### ISSUE TITLE
BigQuery: upgrade to 1.11.0: cannot read from a non-query job 

### ISSUE THREAD

**Issue by pacbeckh** [googleapis/google-cloud-go#2937]
**Client**

BigQuery

**Environment**

Local machine (OSx), docker container: golang:1.13

**Go Environment**

$ go version
go1.15 darwin/amd64

**Code**

```go
query := client.Query(rawQuery)
query.DisableQueryCache = true

rowIterator, err = query.Read(ctx)
```

**Expected behavior**

Run the query

**Actual behavior**

I get the error `cannot read from a non-query job `

**Cause:**
Started after upgrading from 1.9.0 to 1.11.0

Bug seems to be introduced in https://github.com/googleapis/google-cloud-go/pull/2560

The code will poll the job because the response has `JobComplete:false`

But the minimaljob created in the code does not have a config, therefore the `j.isQuery()` check below fails resulting in an error:
```
	if !j.isQuery() {
		return nil, errors.New("bigquery: cannot read from a non-query job")
	}
```
Link to code: 
https://github.com/googleapis/google-cloud-go/blob/26c4a53b2150dc3cfd2e1bb09840072346039aa5/bigquery/job.go#L284

**Comment by shollyman** [comment]
Thanks for the report.  Taking a look.

**Comment by shollyman** [comment]
Will work on releasing a v1.12.0 to pick this up now

**Comment by shollyman** [comment]
I misspoke on the version, this has been released as [bigquery/v1.11.1](https://github.com/googleapis/google-cloud-go/releases/tag/bigquery%2Fv1.11.1) as there were no other significant changes pending.

**Comment by pacbeckh** [comment]
Thanks for addressing this so quickly!
